packages/auth/src/grant/routes.ts

```diff
@@ -84,3 +94,83 @@ async function createGrantInitiation(
   ctx.status = 200
   ctx.body = res
 }
+
+async function startInteraction(
+  deps: ServiceDependencies,
+  ctx: AppContext
+): Promise<void> {
+  const { interactId } = ctx.params
+  const { config, grantService, clientService } = deps
+  const grant = await grantService.getByInteraction(interactId)
+
+  if (!grant) {
+    ctx.status = 401
+    ctx.body = {
+      error: 'unknown_request'
+    }
+
+    return
+  }
+
+  ctx.session.interactId = grant.interactId
+
+  const registryData = await clientService.getRegistryDataByKid(
+    grant.clientKeyId
+  )
+
+  if (!registryData) {
+    ctx.status = 401
+    ctx.body = {
+      error: 'invalid_client'
+    }
+    return
+  }
+
+  const interactionUrl = new URL(config.identityServerDomain)
+  interactionUrl.searchParams.set('clientName', registryData.name)
+  interactionUrl.searchParams.set('clientUri', registryData.url)
+
+  ctx.redirect(interactionUrl.toString())
+}
+
+async function finishInteraction(
+  deps: ServiceDependencies,
+  ctx: AppContext
+): Promise<void> {
+  const { interactId } = ctx.params
+  const interactSession = ctx.session.interactId
+
+  if (!interactSession || !interactId || interactSession !== interactId) {
+    ctx.status = 401
+    ctx.body = {
+      error: 'invalid_request'
+    }
+    return
+  }
+
+  const { grantService, config } = deps
+  const grant = await grantService.getByInteraction(interactId)
+
+  if (!grant) {
+    ctx.status = 404
+    ctx.body = {
+      error: 'unknown_request'
+    }
+    return
+  }
+
+  await grantService.issueGrant(grant.id)
+
+  const clientRedirectUri = new URL(grant.finishUri)
+  const { clientNonce, interactNonce, interactRef } = grant
+  const interactUrl = config.identityServerDomain + `/interact/${interactId}`
+
+  // https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-4.2.3
+  const data = `${clientNonce}\n${interactNonce}\n${interactRef}\n${interactUrl}`
+
+  // TODO: support other hash_methods if provided
+  const hash = crypto.createHash('sha3-512').update(data).digest('base64')
```

Comment:
Same question here.

Reply:
Probably not, I think our spec doesn't offer ways to configure this in a grant request.